In [2]:
from IPython.core.display import display, HTML
display(HTML("<style>.container { width:100% !important; }</style>"))

In [3]:
%%javascript
IPython.OutputArea.prototype._should_scroll = function(lines) {
    return false;
}

<IPython.core.display.Javascript object>

# Occasions Annotations

## Bin of 6+ `guest_count`
Let's look at a few tables:
- `hockey_3_text_processed.csv`
- `silvester_3_text_processed.csv`
- `valentine_3_text_processed.csv`

to get an idea of the different occasions.

###### Our initial set of **occasions** for the category 6+ bin is: 
1. `Drinking` - purpose to drink
    - `Social drinking`
    - `Party`
    
    
2. `Meal` - purpose is to eat
    - `Dinner`
    - `Lunch`
    
    
3. `Social gathring` 
    - `after work` 
    
    
4. `Family Occasion` 

    


### Features:
1. time of day
2. food to drinks ratio
3. kid meal
4. meal step



----------

In [4]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
pd.set_option('display.max_columns', 50)

In [5]:
hockey_path = '../data/hockey_3_text_processed.csv'
silvester_path = '../data/silvester_3_text_processed.csv'
valentine_path = '../data/valentine_3_text_processed.csv'

In [6]:
hockey = pd.read_csv(hockey_path)
silvester = pd.read_csv(silvester_path)
valentine = pd.read_csv(valentine_path)

Fix `datetime` features:

In [7]:
def fix_times(df):
    df['order_time'] = pd.to_datetime(df['order_time'], format="%Y-%m-%d %H:%M:%S.%f")
    df['order_time_closed'] = pd.to_datetime(df['order_time_closed'], format="%Y-%m-%d %H:%M:%S.%f")
    df['order_item_time'] = pd.to_datetime(df['order_item_time'], format="%Y-%m-%d %H:%M:%S.%f")

In [8]:
fix_times(hockey)
fix_times(valentine)
fix_times(silvester)

Lets only look at interesting features

In [9]:
features = ['order_id', 'order_time', 'order_time_closed',
       'order_item_time', 'title', 'category_id', 
       'dwell_time', 'meal_step',
       'meal_flow_last_to_close', 'total_flow_steps', 'meal_flow_step', 
       'beer_volume', 'order_day_of_week',
       'is_weekend', 'guest_count',
       'period_of_day', 
       'sharable', 'kids_meal', 'birthday',
       'total_sales_before_tax', 'item_qty', 'sales_before_tax']

In [10]:
hockey = hockey[features]
silvester = silvester[features]
valentine = valentine[features]

In [11]:
hockey.head()

order_id          order_time   order_time_closed     order_item_time  \
0  520008106 2019-04-16 12:01:00 2019-04-16 13:15:52 2019-04-16 12:13:26   
1  520008106 2019-04-16 12:01:00 2019-04-16 13:15:52 2019-04-16 12:13:26   
2  520008105 2019-04-16 12:01:00 2019-04-16 13:11:55 2019-04-16 12:18:25   
3  520008105 2019-04-16 12:01:00 2019-04-16 13:11:55 2019-04-16 12:13:26   
4  520253565 2019-04-16 19:35:00 2019-04-16 22:39:18 2019-04-16 20:56:13   

               title  category_id  dwell_time  meal_step  \
0  $Sub.Swt.Pot.Fry-          2.0    1.247778          1   
1     FINGERS.SAUCED          2.0    1.247778          1   
2          TRAD.GUAC          2.0    1.181944          2   
3      VEGGIE.BURGER          2.0    1.181944          1   
4        CANADIAN.Pt          1.0    3.071667          2   

   meal_flow_last_to_close  total_flow_steps  meal_flow_step  beer_volume  \
0                       15                18               3        0.000   
1                       15                18               3        0.000   
2                       13                17               4        0.000   
3                       13                17               3        0.000   
4                       12                44              19        0.473   

  order_day_of_week  is_weekend  guest_count period_of_day  sharable  \
0           Tuesday       False            1         lunch         0   
1           Tuesday       False            1         lunch         1   
2           Tuesday       False            1         lunch         1   
3           Tuesday       False            1         lunch         0   
4           Tuesday       False            0        dinner         0   

   kids_meal  birthday  total_sales_before_tax  item_qty  sales_before_tax  
0          0         0                   19.84       1.0              2.97  
1          0         0                   19.84       1.0             16.87  
2          0         0                   26.61       1.0              8.99  
3          0         0                   26.61       1.0             17.62  
4          0         0                   30.48       1.0              7.62

Create table for each order, we're going to pick (randomly) 10 tables out of each dataframe.

In [11]:
def create_tables(df):
    tables = df.drop_duplicates(subset='order_id', keep='first')
    tables.set_index('order_id', inplace=True)
    tables = tables[['guest_count']]

    return tables

In [12]:
hockey_tables = create_tables(hockey)
valentine_tables = create_tables(valentine)
silvester_tables = create_tables(silvester)

In [12]:
print("We have ", len(hockey_tables), " tables in the Hockey data")

We have  118664  tables in the Hockey data


In [13]:
print("We have ", len(valentine_tables), " tables in the Valentine data")

We have  40787  tables in the Valentine data


In [14]:
print("We have ", len(silvester_tables), " tables in the Silvester data")

We have  92107  tables in the Silvester data


In [13]:
hockey.set_index('order_id', inplace=True)
valentine.set_index('order_id', inplace=True)
silvester.set_index('order_id', inplace=True)

-----

# The Annotations

Pick 10 tables out of each dataframe.

In [16]:
COUNT = 10

In [19]:
import random

In [14]:
def get_tables_gc_6(tables):
    return tables[tables['guest_count'] >5]

## Annotating Hockey

Pick 10 tables of a total of 1 guests of **Hockey**.

In [18]:
hockey_6gc_tables = get_tables_gc_6(hockey_tables)
print("We have {} tables of a total of 6+ guests".format(len(hockey_6gc_tables)))

We have 2377 tables of a total of 6+ guests


In [22]:
picked_hockey_tables = random.sample(list(hockey_6gc_tables.index), COUNT)

In [19]:
picked_hockey_tables = [514594544,
 520144566,
 512856753,
 523552965,
 522777307,
 520932269,
 520961529,
 522822209,
 524870243,
 514742701]

# Table 1

In [20]:
#514534188
hockey[hockey.index == picked_hockey_tables[0]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:04:17   
514594544 2019-04-13 19:58:39 2019-04-13 21:23:25 2019-04-13 20:56:00   

                      title  category_id  dwell_time  meal_step  \
order_id                                                          
514594544    Tex Steak Sand          2.0    1.412778          1   
514594544     P.B Deli Sand          2.0    1.412778          1   
514594544        Onion Soup          2.0    1.412778          1   
514594544      Fish & Chips          2.0    1.412778          1   
514594544         Fount Pop          4.0    1.412778          1   
514594544      Fish & Chips          2.0    1.412778          1   
514594544    Tex Steak Sand          2.0    1.412778          1   
514594544     Hot Beef Sand          2.0    1.412778          1   
514594544   Chicken Fingers          2.0    1.412778          1   
514594544         Fount Pop          4.0    1.412778          1   
514594544      Luau Chicken          2.0    1.412778          1   
514594544   Chicken Fingers          2.0    1.412778          1   
514594544   Matzo Ball Soup          2.0    1.412778          1   
514594544   Matzo Ball Soup          2.0    1.412778          1   
514594544        Cafe Latte          4.0    1.412778          1   
514594544         Fount Pop          4.0    1.412778          1   
514594544   Deli Meat Plate          2.0    1.412778          1   
514594544         Fount Pop          4.0    1.412778          1   
514594544   Matzo Ball Soup          2.0    1.412778          1   
514594544  SAT- 14oz Stella          1.0    1.412778          1   
514594544            Coffee          4.0    1.412778          2   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
514594544                        6                19               1   
514594544                        6                19               1   
514594544                        6                19               1   
514594544                        6                19               1   
514594544                        6                19               1   
514594544                        6                19               1   
514594544                        6                19               1   
514594544                        6                19               1   
514594544                        6                19               1   
51459

# **April 13th (Saturday), 19:58 - 21:23, a total of 1.5 hours**

**Meal Steps Description:**
1. Start
  - Meal very heavy, all ordered at once

2. Coffe 30 min later
  - Dessert

      
**`Dinner`** 


**Explanation**: Because of the time and the size of the meal.

**Features that helped**: the amount of foold on the table is approxemtly the GC, `price of meal` (10 $), and `time of meal` (21:00pm). 

# Table 2

In [21]:
hockey[hockey.index == picked_hockey_tables[1]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 19:51:43   
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 19:51:43   
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 19:51:43   
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 20:00:32   
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 20:00:32   
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 20:00:32   
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 20:00:32   
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 20:00:32   
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 20:00:32   
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 20:00:32   
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 20:00:32   
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 20:00:32   
520144566 2019-04-16 19:51:09 2019-04-16 20:37:33 2019-04-16 20:34:59   

                     title  category_id  dwell_time  meal_step  \
order_id                                                         
520144566    Deluxe Dinner          2.0    0.773333          1   
520144566    Deluxe Dinner          2.0    0.773333          1   
520144566    Deluxe Dinner          2.0    0.773333          1   
520144566              Pop          4.0    0.773333          2   
520144566            Evian          4.0    0.773333          2   
520144566         Guinness          1.0    0.773333          2   
520144566         Heineken          1.0    0.773333          2   
520144566  Castle Cab Sauv          6.0    0.773333          2   
520144566   San Pellegrino          4.0    0.773333          2   
520144566  Mill St Organic          1.0    0.773333          2   
520144566       LC Crimzen          1.0    0.773333          2   
520144566           921 PG          6.0    0.773333          2   
520144566       Room Setup          5.0    0.773333          2   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
520144566                        0                10               0   
520144566                        0                10               0   
520144566                        0                10               0   
520144566                        0                10               2   
520144566                        0                10               2   
520144566                        0                10               2   
520144566                        0                10               2   
520144566                        0                10               2   
520144566                        0                10               2   
520144566                        0                10               2   
520144566                        0                10               2   
520144566                        0                10               2   
520144566                        0                10              10   

           beer_volume order_day_of_week  is_weekend  guest_count  \
order_id                                                            
520144566       -0.000           Tuesday       False           22   
520144566        0.000           Tuesday       False           22   
520144566       -0.000           Tuesday       False           22   
520144566        0.000           Tuesday       False           22   
520144566        0.000           Tuesday       False           22   
520144566        0.440           Tuesday       False           22   
520144566        0.660           Tuesday       False           22   
520144566        0.000           Tuesday       False           22   
520144566        0.000           Tuesday       False           22   
520144566        0.682           Tuesday       False           22   
520144566        0.473 

**April 16th (Tuesday), 19:51 - 20:37, total 0.77 hours, 2 meal steps**

** 22 people eat dinner and some order more soft drinks and liquer 
I dont use the the dwell time here
**Meal Steps Description:**
it actually has a single meal flow and not 2 as mentioned. 
       
**Total drinks**: 5 alcholic drinks 3 soft drinks

**Dinner Party**

**Explanation**: This is a party where the foucus is on the food, room was set for the group. 

**Features**: number of dinners which was served for the table, amount of food over the amount of drinks

rule: if the guest count is greather than 12 then it was set before, therefore it a big event and with high ratio of food over drinks it is a dinner event. 

# Table 3

In [28]:
#521093892
hockey[hockey.index == picked_hockey_tables[2]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
512856753 2019-04-11 20:35:00 2019-04-12 01:25:07 2019-04-11 20:35:00   
512856753 2019-04-11 20:35:00 2019-04-12 01:25:07 2019-04-11 20:41:00   
512856753 2019-04-11 20:35:00 2019-04-12 01:25:07 2019-04-11 21:21:00   
512856753 2019-04-11 20:35:00 2019-04-12 01:25:07 2019-04-11 22:31:00   
512856753 2019-04-11 20:35:00 2019-04-12 01:25:07 2019-04-11 23:27:00   
512856753 2019-04-11 20:35:00 2019-04-12 01:25:07 2019-04-12 00:28:00   
512856753 2019-04-11 20:35:00 2019-04-12 01:25:07 2019-04-12 00:37:00   
512856753 2019-04-11 20:35:00 2019-04-12 01:25:07 2019-04-12 00:37:00   
512856753 2019-04-11 20:35:00 2019-04-12 01:25:07 2019-04-12 00:52:00   

                      title  category_id  dwell_time  meal_step  \
order_id                                                          
512856753    LG COORS LIGHT          1.0    4.835278          1   
512856753  CHICKEN STIR FRY          2.0    4.835278          2   
512856753    LG COORS LIGHT          1.0    4.835278          2   
512856753    LG COORS LIGHT          1.0    4.835278          2   
512856753    LG COORS LIGHT          1.0    4.835278          2   
512856753    LG COORS LIGHT          1.0    4.835278          2   
512856753    LG COORS LIGHT          1.0    4.835278          2   
512856753    LG COORS LIGHT          1.0    4.835278          2   
512856753     PEPPERONI SML          2.0    4.835278          2   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
512856753                        8                70               0   
512856753                        8                70               1   
512856753                        8                70              11   
512856753                        8                70              28   
512856753                        8                70              42   
512856753                        8                70              57   
512856753                        8                70              59   
512856753                        8                70              59   
512856753                        8                70              62   

           beer_volume order_day_of_week  is_weekend  guest_count  \
order_id                                                            
512856753        0.591          Thursday       False           11   
512856753        0.000          Thursday       False           11   
512856753        0.591          Thursday       False           11   
512856753        0.591          Thursday       False           11   
512856753        0.591          Thursday       False           11   
512856753        0.591          Thursday       False           11   
512856753        0.591          Thursday       False           11   
512856753        0.591          Thursday       False           11   
512856753        0.000          Thursday       False           11   

          period_of_day  sharable  kids_meal  birthday  \
order_id                                                 
512856753        dinner         0          0         0   
512856753        dinner         0          0         0   
512856753        dinner         0          0         0   
512856753        dinner         0          0         0   
512856753        dinner         0          0         0   
512856753        dinner         0          0         0   
512856753        dinner         0          0         0   
512856753        dinner         0          0         0   
512856753        dinner         0          0         0   

           total_sales_before_tax  item_qty  sales_before_tax  
order_id                                                       
512856753                   76.75       1.0              7.25  
512856753                   76.75       1.0             16.00  
512856753                   7

**Social Gathering**

**April 11th (Thursday), 20:35 - 01:25, total of 4.83 hours**

**Meal Steps Description:**
1. food and beer
2. 40 minutes later: beer
3. 70 minutes later: beer
4. 60 min later: beer
5. 60 min later: 2 beer
6. 15 min later: pizza


**`Wrong Label`**


**Explanation**: Its hard to say how many people are on the table but it is less than 11 and probabily less than 6 so I classifity this as a wornk label of GC



# Table 4

In [29]:
hockey[hockey.index == picked_hockey_tables[3]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 18:42:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 18:42:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 18:50:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 18:50:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 19:04:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 19:04:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 19:04:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 19:04:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 19:04:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 19:04:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 19:28:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 20:03:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 20:03:00   
523552965 2019-04-20 18:42:00 2019-04-20 20:31:50 2019-04-20 20:03:00   

                   title  category_id  dwell_time  meal_step  \
order_id                                                       
523552965           SODA          4.0    1.830556          1   
523552965           COKE          4.0    1.830556          1   
523552965       CALAMARI          2.0    1.830556          2   
523552965       CALAMARI          2.0    1.830556          2   
523552965  CHICKN FAJITA          2.0    1.830556          2   
523552965  HALIBUT&CHIPS          2.0    1.830556          2   
523552965   STEAK FAJITA          2.0    1.830556          2   
523552965   STEAK FAJITA          2.0    1.830556          2   
523552965      2LB WINGS          2.0    1.830556          2   
523552965   KIDS-ALFREDO          2.0    1.830556          2   
523552965       HEINEKEN          1.0    1.830556          2   
523552965   FUDGE SUNDAE          2.0    1.830556          2   
523552965  BUTTER SUNDAE          2.0    1.830556          2   
523552965  STRAW CH CAKE          2.0    1.830556          2   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
523552965                        7                26               0   
523552965                        7                26               0   
523552965                        7                26               2   
523552965                        7                26               2   
523552965                        7                26               5   
523552965                        7                26               5   
523552965                        7                26               5   
523552965                        7                26               5   
523552965                        7                26               5   
523552965                        7                26               5   
523552965                        7                26              11   
523552965                        7                26              19   
523552965                        7                26              19   
523552965                        7                26              19   

           beer_volume order_day_of_week  is_weekend  guest_count  \
order_id                                                            
523552965         0.00          Saturday        True            6   
523552965         0.00          Saturday        True            6   
523552965         0.00          Saturday        True            6   
523552965         0.00          Saturday        True            6   
523552965         0.00          Saturday        True            6   
523552965         0.00          Saturday        True            6   
523552965         0.00          Saturday        True            6   
523552965         0.00          Saturday        Tru

**`Dinner`**
Family dinner 


There is also a "kid" meal in here.

many foods and single beer with two soft drinks

**Features**: GC size of a family, rator of food to alcohol beverage is high. small number of soft drinks

# Table 5

In [30]:
hockey[hockey.index == picked_hockey_tables[4]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
522777307 2019-04-20 00:06:00 2019-04-20 00:45:46 2019-04-20 00:06:32   
522777307 2019-04-20 00:06:00 2019-04-20 00:45:46 2019-04-20 00:06:32   
522777307 2019-04-20 00:06:00 2019-04-20 00:45:46 2019-04-20 00:06:32   
522777307 2019-04-20 00:06:00 2019-04-20 00:45:46 2019-04-20 00:06:32   
522777307 2019-04-20 00:06:00 2019-04-20 00:45:46 2019-04-20 00:06:32   
522777307 2019-04-20 00:06:00 2019-04-20 00:45:46 2019-04-20 00:06:32   
522777307 2019-04-20 00:06:00 2019-04-20 00:45:46 2019-04-20 00:24:55   
522777307 2019-04-20 00:06:00 2019-04-20 00:45:46 2019-04-20 00:24:55   
522777307 2019-04-20 00:06:00 2019-04-20 00:45:46 2019-04-20 00:24:55   
522777307 2019-04-20 00:06:00 2019-04-20 00:45:46 2019-04-20 00:24:55   
522777307 2019-04-20 00:06:00 2019-04-20 00:45:46 2019-04-20 00:34:47   

                      title  category_id  dwell_time  meal_step  \
order_id                                                          
522777307  JACK DANIELS DBL          3.0    0.662778          1   
522777307       CUERVO GOLD          3.0    0.662778          1   
522777307            CORONA          1.0    0.662778          1   
522777307            CORONA          1.0    0.662778          1   
522777307          RED BULL          4.0    0.662778          1   
522777307      HENNESSY DBL          3.0    0.662778          1   
522777307     BURT REYNOLDS          3.0    0.662778          2   
522777307            CORONA          1.0    0.662778          2   
522777307      HENNESSY DBL          3.0    0.662778          2   
522777307      HENNESSY DBL          3.0    0.662778          2   
522777307          RED BULL          4.0    0.662778          2   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
522777307                        2                 8               0   
522777307                        2                 8               0   
522777307                        2                 8               0   
522777307                        2                 8               0   
522777307                        2                 8               0   
522777307                        2                 8               0   
522777307                        2                 8               4   
522777307                        2                 8               4   
522777307                        2                 8               4   
522777307                        2                 8               4   
522777307                        2                 8               6   

           beer_volume order_day_of_week  is_weekend  guest_count  \
order_id                                                            
522777307        0.000          Saturday        True            6   
522777307        0.000          Saturday        True            6   
522777307        0.355          Saturday        True            6   
522777307        0.355          Saturday        True            6   
522777307        0.000          Saturday        True            6   
522777307        0.000          Saturday        True            6   
522777307        0.000          Saturday        True            6   
522777307        0.355          Saturday        True            6   
522777307        0.000          Saturday        True            6   
522777307        0.000          Saturday        True            6   
522777307        0.000          Saturday        True            6   

          period_of_day  sharable  kids_meal  birthday  \
order_id                                                 
522777307         night         0          0         0   
522777307         night         0          0         0   
522777307         night         0          0         0   
522777307         night         0          0         0   
522777307     

**April 20th (Sauturday), 00:06 - 00:45, total of 0.66 hours**

**`drinking`**

only drinks on the table, 2 meal step and with six rows per meal step. 


**Explanation**: Not much to explain, a group of people are drinking.


# Table 6

In [31]:
hockey[hockey.index == picked_hockey_tables[5]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:20:51   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:20:51   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:28:38   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:28:38   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:29:39   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:30:17   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:30:17   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:33:39   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:41:33   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:41:33   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:41:33   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:41:33   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:41:33   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:41:33   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:41:33   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 17:41:33   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 18:02:36   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 18:02:36   
520932269 2019-04-17 17:20:00 2019-04-17 18:43:31 2019-04-17 18:08:42   

                       title  category_id  dwell_time  meal_step  \
order_id                                                           
520932269               POP.          4.0    1.391944          1   
520932269        CREEMORE.Pt          1.0    1.391944          1   
520932269               POP.          4.0    1.391944          2   
520932269               POP.          4.0    1.391944          2   
520932269        Loaded.Mash          2.0    1.391944          1   
520932269   SAM.ADAM.SEAS.Pt          1.0    1.391944          1   
520932269   BELGIAN.MOON..Pt          1.0    1.391944          1   
520932269          TUNA.GUAC          2.0    1.391944          1   
520932269     FINGERS.SAUCED          2.0    1.391944          2   
520932269             Mashed          2.0    1.391944          2   
520932269        Loaded.Mash          2.0    1.391944          2   
520932269             Mashed          2.0    1.391944          2   
520932269      VEGGIE.BURGER          2.0    1.391944          2   
520932269       cajun.shrimp          2.0    1.391944          2   
520932269  $$Truffle.Fries$$          2.0    1.391944          2   
520932269         Side.House          2.0    1.391944          2   
520932269        CREEMORE.XL          1.0    1.391944          2   
520932269             WATER.          4.0    1.391944          2   
520932269   BELGIAN.MOON..Pt          1.0    1.391944          2   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
520932269                        8                16               0   
520932269                        8                16               0   
520932269                        8                16               1   
520932269                        8                16               1   
520932269                        8                16               1   
520932269                        8                16               1   
520932269                        8                16               1   
520932269                        8                16               1   
520932269                        8                16               2   
520932269                        8                16               2   
520932269                        8                16               2   
520932269                        8                16               2   
520932269                        8              

**April 17th (Wednesday), 17:20 - 18:43, total of 1.39 hours,.**

**`Dinner`**

**Explanation**: different types of food and varity of drinks, seems like a grown up group gathering after work.
**Features**:  day of week, hour, ration of food to drinks is almost 1



# Table 7

In [29]:
hockey[hockey.index == picked_hockey_tables[6]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:22:01   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:22:01   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:22:01   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:22:01   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:22:01   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:22:01   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:22:27   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:22:27   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:27:34   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:27:53   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:27:53   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:28:11   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:30:40   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:30:40   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:30:40   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:30:40   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:30:40   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:30:40   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:50:41   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:50:41   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:50:41   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:50:41   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:50:41   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 18:50:41   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 19:01:07   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 19:13:05   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 19:18:48   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 19:27:36   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 19:27:36   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 19:27:36   
520961529 2019-04-17 18:22:00 2019-04-17 20:00:21 2019-04-17 19:27:36   

                     title  category_id  dwell_time  meal_step  \
order_id                                                         
520961529   RICKARD.RED.Pt          1.0    1.639167          1   
520961529      HEINEKEN.Pt          1.0    1.639167          1   
520961529           KIDBEV          4.0    1.639167          1   
520961529           KIDBEV          4.0    1.639167          1   
520961529           KIDBEV          4.0    1.639167          1   
520961529             POP.          4.0    1.639167          1   
520961529           KIDBEV          4.0    1.639167          1   
520961529           WATER.          4.0    1.639167          1   
520961529   RICKARD.RED.Pt          1.0    1.639167          2   
520961529       WINGS.15PC          2.0    1.639167          1   
520961529            WINGS          2.0    1.639167          1   
520961529         CALAMARI          2.0    1.639167          1   
520961529     K.PEPP.PIZZA          2.0    1.639167          1   
520961529      K.CHK.BITES          2.0    1.639167          1   
520961529          K.PIZZA          2.0    1.639167          1   
520961529          K.PIZZA          2.0    1.639167          1   
520961529          K.PIZZA          2.0    1.639167          1   
520961529            k.veg          2.0    1.639167          1   
520961529   Chikn.Rice/Veg          2.0    1.639167          2   
520961529   DBL.BAC.CHSBRG          2.0    1.639167          2   
520961529      DBL.CHSBURG          2.0    1.639167          2   
520961529  Salmon.Rice/Veg          2.0    1.639167          2   
520961529     ROAST.CHICK

**April 17th (Wednesday), 18:22 - 20:00, total of 1.63 hours**

**`Family`**        

**Explanation**: Its a classic dinner, maybe also watchign a game or something. Starts with a drinks and order food for many people.later order some desserts.


**Features**: staying at the table for 2 hours, many drinks and many food, ratio is almost 0.5  

# Table 8

In [28]:
hockey[hockey.index == picked_hockey_tables[7]].sort_values(by=['order_item_time'])

NameError: name 'picked_hockey_tables' is not defined

**April 20nd (Saturday), 01:48 - 02:18, total of 0.5 hours**

       
**total drinks**: 4 beers and 10 liquers,

**`Social drinking`**

**Explanation**: Its a drinking table, Its a late night, they came to drink.

**Features**: staying at the table for 30 minutes, food to drink ratio is 0. many drinks, hard liquer

# Valentine

Pick 5 tables of total of 1 guests for **Valentine**.

In [24]:
valentine_gc = get_tables_gc_6(valentine_tables)

print("We have {} tables of a total of 1 guests".format(len(valentine_gc)))

We have 720 tables of a total of 1 guests


In [26]:
import random
picked_valentine_tables = random.sample(list(valentine_gc.index), COUNT)

In [28]:
picked_valentine_tables = [434693771,
 446750517,
 447075894,
 447589967,
 434539751,
 448045237,
 448730983,
 447983682,
 447100562,
 434734939]

## Table 1

In [29]:
valentine[valentine.index == picked_valentine_tables[0]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 17:38:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 17:51:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:04:00   
434693771 2019-02-10 17:38:00 2019-02-10 19:46:28 2019-02-10 18:29:00   

                      title  category_id  dwell_time  meal_step  \
order_id                                                          
434693771  W.KIM.CRAW.SV.1z          6.0    2.141111          1   
434693771  R.ST.JEAN.CAB.Bt          6.0    2.141111          2   
434693771       7OZ.SIRLOIN          2.0    2.141111          3   
434693771      BACON.FILET*          2.0    2.141111          3   
434693771       LAMB.SALAD*          2.0    2.141111          3   
434693771  $Truffle.Fries$$          2.0    2.141111          3   
434693771        CAESAR.APP          2.0    2.141111          3   
434693771        TUNA.STACK          2.0    2.141111          3   
434693771          b.potato          2.0    2.141111          3   
434693771    PEPPERCORN.NY*          2.0    2.141111          3   
434693771    10OZ.PRIME.RIB          2.0    2.141111          3   
434693771     AVOCADO.SALAD          2.0    2.141111          3   
434693771    K.C.STEAK.SOUP          2.0    2.141111          3   
434693771      BACON.FILET*          2.0    2.141111          3   
434693771       7OZ.SIRLOIN          2.0    2.141111          3   
434693771  W.KIM.CRAW.SV.1z          6.0    2.141111          4   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
434693771                       19                31               0   
434693771                       19                31               3   
434693771                       19                31               6   
434693771                       19                31               6   
434693771                       19                31               6   
434693771                       19                31               6   
434693771                       19                31               6   
434693771                       19                31               6   
434693771                       19                31               6   
434693771                       19                31               6   
434693771                       19                31               6   
434693771                       19                31               6   
434693771                       19                31               6   
434693771                       19                31               6   
434693771                       19                31               6   
434693771                       19                31              12   

           beer_volume order_day_of_week  is_weekend  guest_count  \
order_id                                                            
434693771          0.0            Sunday        True          

**Feb 10th (Sunday), 17:38 - 19:46, total of ~2 hour, 5 meal steps.**

**total drinks**: 3 wines,

**`family dinner`**


# Table 2

In [30]:
valentine[valentine.index == picked_valentine_tables[1]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
446750517 2019-02-12 12:07:00 2019-02-12 13:09:18 2019-02-12 12:07:00   
446750517 2019-02-12 12:07:00 2019-02-12 13:09:18 2019-02-12 12:07:00   
446750517 2019-02-12 12:07:00 2019-02-12 13:09:18 2019-02-12 12:18:00   
446750517 2019-02-12 12:07:00 2019-02-12 13:09:18 2019-02-12 12:18:00   
446750517 2019-02-12 12:07:00 2019-02-12 13:09:18 2019-02-12 12:32:00   
446750517 2019-02-12 12:07:00 2019-02-12 13:09:18 2019-02-12 12:45:00   

                      title  category_id  dwell_time  meal_step  \
order_id                                                          
446750517  LRG BELGIAN MOON          1.0    1.038333          1   
446750517          HEINEKEN          1.0    1.038333          1   
446750517      CHEF SPECIAL          2.0    1.038333          2   
446750517    CHKN CLUBHOUSE          2.0    1.038333          2   
446750517          HEINEKEN          1.0    1.038333          3   
446750517        RG SLEEMAN          1.0    1.038333          4   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
446750517                        6                14               0   
446750517                        6                14               0   
446750517                        6                14               2   
446750517                        6                14               2   
446750517                        6                14               5   
446750517                        6                14               8   

           beer_volume order_day_of_week  is_weekend  guest_count  \
order_id                                                            
446750517        0.591           Tuesday       False            8   
446750517        0.330           Tuesday       False            8   
446750517        0.000           Tuesday       False            8   
446750517        0.000           Tuesday       False            8   
446750517        0.330           Tuesday       False            8   
446750517        0.355           Tuesday       False            8   

          period_of_day  sharable  kids_meal  birthday  \
order_id                                                 
446750517         lunch         0          0         0   
446750517         lunch         0          0         0   
446750517         lunch         0          0         0   
446750517         lunch         0          0         0   
446750517         lunch         0          0         0   
446750517         lunch         0          0         0   

           total_sales_before_tax  item_qty  sales_before_tax  
order_id                                                       
446750517                   57.85       1.0              7.25  
446750517                   57.85       1.0              7.20  
446750517                   57.85       1.0             16.97  
446750517                   57.85       1.0             15.00  
446750517                   57.85       1.0              7.20  
446750517                   57.85       1.0              4.23

**Feb 12th (Tuesday), 12:07 - 13:09, total of 1 hour , 4 meal steps.**


       
**total meals**: 2 foods 4 beers

wrong label






# Table 3

In [31]:
valentine[valentine.index == picked_valentine_tables[2]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:41:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:41:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:41:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:41:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:41:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:41:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:41:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:41:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:41:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:56:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:56:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:56:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:56:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:56:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:56:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:56:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:56:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:56:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:56:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 18:56:00   
447075894 2019-02-13 18:41:00 2019-02-13 19:52:42 2019-02-13 19:03:00   

                       title  category_id  dwell_time  meal_step  \
order_id                                                           
447075894               POP.          4.0       1.195          1   
447075894   BLURASP LEMONADE          3.0       1.195          1   
447075894      BAR.MANGER.XL          1.0       1.195          1   
447075894   DBL.BARRL.SHZ.Lg          6.0       1.195          1   
447075894   BLURASP LEMONADE          3.0       1.195          1   
447075894      MARGRITA.LIME          3.0       1.195          1   
447075894       SML.RED.SANG          3.0       1.195          1   
447075894          STELLA.Pt          1.0       1.195          1   
447075894            BEACHED          3.0       1.195          1   
447075894   CHEESE.CHIKN.MAC          2.0       1.195          2   
447075894     SESAME CHICKEN          2.0       1.195          2   
447075894   SUB.POUT.CLASSIC          5.0       1.195          2   
447075894  PAD.THAI.CHK/SHMP          2.0       1.195          2   
447075894    CHICKEN.FINGERS          2.0       1.195          2   
447075894   CHEESE.CHIKN.MAC          2.0       1.195          2   
447075894             Mashed          2.0       1.195          2   
447075894             Mashed          2.0       1.195          2   
447075894   CHEESE.CHIKN.MAC          2.0       1.195          2   
447075894           PAD THAI          2.0       1.195          2   
447075894     DBL.PLAIN.BURG          2.0       1.195          2   
447075894    BREWSKY.RT.BEER          4.0       1.195          3   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
447075894                       12                16               0   
447075894                       12                16               0   
447075894                       12                16               0   
447075894                       12                16               0   
447075894                       12                16               0   
447075894                       12                16               0   
447075894                       12                16               0   
447075894                       12                16               0   
447075894                       12                16  

**Feb 13th (Wednesday), 13:08 - 14:07, total of ~1 hour, 3 meal steps.**

**total drinks**: 10 
**total food**: 9

**`Dinner`**

**Explanation**: classic Dinner. 

**Features**: time of meal , and spent about an hour.

# Table 4

In [32]:
valentine[valentine.index == picked_valentine_tables[3]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
447589967 2019-02-14 17:46:00 2019-02-14 20:59:33 2019-02-14 17:46:00   
447589967 2019-02-14 17:46:00 2019-02-14 20:59:33 2019-02-14 17:46:00   
447589967 2019-02-14 17:46:00 2019-02-14 20:59:33 2019-02-14 17:46:00   
447589967 2019-02-14 17:46:00 2019-02-14 20:59:33 2019-02-14 17:46:00   
447589967 2019-02-14 17:46:00 2019-02-14 20:59:33 2019-02-14 17:55:00   
447589967 2019-02-14 17:46:00 2019-02-14 20:59:33 2019-02-14 17:55:00   
447589967 2019-02-14 17:46:00 2019-02-14 20:59:33 2019-02-14 17:55:00   
447589967 2019-02-14 17:46:00 2019-02-14 20:59:33 2019-02-14 17:55:00   
447589967 2019-02-14 17:46:00 2019-02-14 20:59:33 2019-02-14 17:55:00   
447589967 2019-02-14 17:46:00 2019-02-14 20:59:33 2019-02-14 17:55:00   

                       title  category_id  dwell_time  meal_step  \
order_id                                                           
447589967             STELLA          1.0    3.225833          1   
447589967       LRG RICKARDS          1.0    3.225833          1   
447589967   LRG BELGIAN MOON          1.0    3.225833          1   
447589967      1/2 LITRE RED          6.0    3.225833          1   
447589967        STEAK BITES          2.0    3.225833          2   
447589967        EIGHT WINGS          2.0    3.225833          2   
447589967               1.50          2.0    3.225833          2   
447589967  CAULIFLOWER BITES          2.0    3.225833          2   
447589967    SAUSAGE MARGESE          2.0    3.225833          2   
447589967     NACHOS SUPREME          2.0    3.225833          2   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
447589967                       46                48               0   
447589967                       46                48               0   
447589967                       46                48               0   
447589967                       46                48               0   
447589967                       46                48               2   
447589967                       46                48               2   
447589967                       46                48               2   
447589967                       46                48               2   
447589967                       46                48               2   
447589967                       46                48               2   

           beer_volume order_day_of_week  is_weekend  guest_count  \
order_id                                                            
447589967        0.330          Thursday       False            6   
447589967        0.591          Thursday       False            6   
447589967        0.591          Thursday       False            6   
447589967        0.000          Thursday       False            6   
447589967        0.000          Thursday       False            6   
447589967        0.000          Thursday       False            6   
447589967        0.000          Thursday       False            6   
447589967        0.000          Thursday       False            6   
447589967        0.000          Thursday       False            6   
447589967        0.000          Thursday       False            6   

          period_of_day  sharable  kids_meal  birthday  \
order_id                                                 
447589967     afternoon         0          0         0   
447589967     afternoon         0          0         0   
447589967     afternoon         0          0         0   
447589967     afternoon         0          0         0   
447589967     afternoon         0          0         0   
447589967     afternoon         1          0         0   
447589967     afternoon         0          0         0   
447589967     afternoon         0          0         0   
447589967     afternoon         0          0       

**`dinner`**

food to drinks ration is higer than 1

time of day

total amout of orders

# Silvester

In [18]:
import random
silvester_gc = get_tables_gc_6(silvester_tables)
print("We have {} tables of a total of 1 guests".format(len(silvester_gc)))

We have 2190 tables of a total of 1 guests


In [20]:
picked_silvester_tables = random.sample(list(silvester_gc.index), 3)

In [22]:
picked_silvester_tables = [361348245, 360970732, 360903046]

# Table 1

In [33]:
silvester[silvester.index == picked_silvester_tables[0]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
361348245 2018-12-29 19:05:31 2018-12-29 19:55:18 2018-12-29 19:09:13   
361348245 2018-12-29 19:05:31 2018-12-29 19:55:18 2018-12-29 19:09:13   
361348245 2018-12-29 19:05:31 2018-12-29 19:55:18 2018-12-29 19:09:13   
361348245 2018-12-29 19:05:31 2018-12-29 19:55:18 2018-12-29 19:09:13   
361348245 2018-12-29 19:05:31 2018-12-29 19:55:18 2018-12-29 19:09:13   
361348245 2018-12-29 19:05:31 2018-12-29 19:55:18 2018-12-29 19:09:13   
361348245 2018-12-29 19:05:31 2018-12-29 19:55:18 2018-12-29 19:09:13   
361348245 2018-12-29 19:05:31 2018-12-29 19:55:18 2018-12-29 19:09:13   
361348245 2018-12-29 19:05:31 2018-12-29 19:55:18 2018-12-29 19:09:13   
361348245 2018-12-29 19:05:31 2018-12-29 19:55:18 2018-12-29 19:09:13   
361348245 2018-12-29 19:05:31 2018-12-29 19:55:18 2018-12-29 19:09:13   
361348245 2018-12-29 19:05:31 2018-12-29 19:55:18 2018-12-29 19:09:13   

                      title  category_id  dwell_time  meal_step  \
order_id                                                          
361348245   Fettuccine Pomo          2.0    0.829722          1   
361348245    3 Alarm Burger          2.0    0.829722          1   
361348245   PB House Burger          2.0    0.829722          1   
361348245      Great Reuben          5.0    0.829722          1   
361348245    Gnocchi Tomato          2.0    0.829722          1   
361348245  S-Taco Baja Fish          2.0    0.829722          1   
361348245    Mexican Shrimp          2.0    0.829722          1   
361348245   Matzo Ball Soup          2.0    0.829722          1   
361348245     P.B Deli Sand          2.0    0.829722          1   
361348245  Chicken Parmesan          2.0    0.829722          1   
361348245  Seafood Linguine          2.0    0.829722          1   
361348245    Gnocchi Tomato          2.0    0.829722          1   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
361348245                       11                11               0   
361348245                       11                11               0   
361348245                       11                11               0   
361348245                       11                11               0   
361348245                       11                11               0   
361348245                       11                11               0   
361348245                       11                11               0   
361348245                       11                11               0   
361348245                       11                11               0   
361348245                       11                11               0   
361348245                       11                11               0   
361348245                       11                11               0   

           beer_volume order_day_of_week  is_weekend  guest_count  \
order_id                                                            
361348245          0.0          Saturday        True            9   
361348245          0.0          Saturday        True            9   
361348245          0.0          Saturday        True            9   
361348245          0.0          Saturday        True            9   
361348245          0.0          Saturday        True            9   
361348245          0.0          Saturday        True            9   
361348245          0.0          Saturday        True            9   
361348245          0.0          Saturday        True            9   
361348245          0.0          Saturday        True            9   
361348245          0.0          Saturday        True            9   
361348245          0.0          Saturday        True            9   
361348245         -0.0          Saturday        True            9   

          period_of_day  sharable  kids_meal  birthday  \
order_id               

**DEC 29th (Saturday), 19:05 - 19:55, total of 50 minutes, 1 meal steps.**

**`dinner`** 

`dinner_time`, `dewll_meal`


# Table 2

In [25]:
silvester[silvester.index == picked_silvester_tables[1]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
360970732 2018-12-28 21:39:51 2018-12-28 23:01:16 2018-12-28 21:40:10   
360970732 2018-12-28 21:39:51 2018-12-28 23:01:16 2018-12-28 21:40:10   
360970732 2018-12-28 21:39:51 2018-12-28 23:01:16 2018-12-28 21:49:23   
360970732 2018-12-28 21:39:51 2018-12-28 23:01:16 2018-12-28 21:49:23   
360970732 2018-12-28 21:39:51 2018-12-28 23:01:16 2018-12-28 21:49:23   
360970732 2018-12-28 21:39:51 2018-12-28 23:01:16 2018-12-28 21:49:23   
360970732 2018-12-28 21:39:51 2018-12-28 23:01:16 2018-12-28 21:49:23   
360970732 2018-12-28 21:39:51 2018-12-28 23:01:16 2018-12-28 21:49:23   
360970732 2018-12-28 21:39:51 2018-12-28 23:01:16 2018-12-28 22:03:36   
360970732 2018-12-28 21:39:51 2018-12-28 23:01:16 2018-12-28 22:03:36   
360970732 2018-12-28 21:39:51 2018-12-28 23:01:16 2018-12-28 22:35:35   
360970732 2018-12-28 21:39:51 2018-12-28 23:01:16 2018-12-28 22:35:35   

                      title  category_id  dwell_time  meal_step  \
order_id                                                          
360970732     Green Monster          4.0    1.356944          1   
360970732      Killer Shake          4.0    1.356944          1   
360970732  Maple Nut Salmon          2.0    1.356944          2   
360970732   Fettuccine Pomo          2.0    1.356944          2   
360970732  Spag & Meatballs          2.0    1.356944          2   
360970732  K- Spag&Meatball          2.0    1.356944          2   
360970732  S-Mac&Chz Popprs          2.0    1.356944          2   
360970732  Maple Nut Salmon          2.0    1.356944          2   
360970732  S-Moroccan Cigrs          2.0    1.356944          3   
360970732  K- Spag&Meatball          2.0    1.356944          3   
360970732   Chzcake WhtChoc          2.0    1.356944          4   
360970732     ChzCake Plain          2.0    1.356944          4   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
360970732                        6                18               0   
360970732                        6                18               0   
360970732                        6                18               2   
360970732                        6                18               2   
360970732                        6                18               2   
360970732                        6                18               2   
360970732                        6                18               2   
360970732                        6                18               2   
360970732                        6                18               5   
360970732                        6                18               5   
360970732                        6                18              12   
360970732                        6                18              12   

           beer_volume order_day_of_week  is_weekend  guest_count  \
order_id                                                            
360970732          0.0            Friday        True            6   
360970732          0.0            Friday        True            6   
360970732          0.0            Friday        True            6   
360970732          0.0            Friday        True            6   
360970732          0.0            Friday        True            6   
360970732          0.0            Friday        True            6   
360970732          0.0            Friday        True            6   
360970732          0.0            Friday        True            6   
360970732          0.0            Friday        True            6   
360970732          0.0            Friday        True            6   
360970732          0.0            Friday        True            6   
360970732          0.0            Friday        True            6   

          period_of_day  sharable  kids_meal  birthday  \
order_id               

**Dec 28th (Friday), 21:39 - 23:01, 1.35 hrs, 4 meal steps.**

**Total drinks**: No Drinks

**`Family Occasion`** 

late night, no drinks, K dish



# Table 3

In [26]:
silvester[silvester.index == picked_silvester_tables[2]].sort_values(by=['order_item_time'])

order_time   order_time_closed     order_item_time  \
order_id                                                                
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   
360903046 2018-12-28 19:25:00 2018-12-28 20:46:29 2018-12-28 19:30:00   

                       title  category_id  dwell_time  meal_step  \
order_id                                                           
360903046        CRAB.CAKES*          2.0    1.358056          1   
360903046      PF-CHEESECAKE          2.0    1.358056          1   
360903046      PF-CHEESECAKE          2.0    1.358056          1   
360903046     CHICKEN.CAESAR          2.0    1.358056          1   
360903046   R.BER.CAL.CAB.1z          6.0    1.358056          1   
360903046         GRILL.RIBS          2.0    1.358056          1   
360903046    PF-GARDEN.SALAD          2.0    1.358056          1   
360903046     PF-PAN.SALMON.          2.0    1.358056          1   
360903046    CHKN.COBB.SALAD          2.0    1.358056          1   
360903046   PF-SHRMP.RAVIOLI          2.0    1.358056          1   
360903046     PF-SEASON.SOUP          2.0    1.358056          1   
360903046    PF-CREME.BRULEE          2.0    1.358056          1   
360903046  PF-10oz.PRIME.RIB          2.0    1.358056          1   
360903046     PF-SEASON.SOUP          2.0    1.358056          1   
360903046    SHRIMP.COCKTAIL          2.0    1.358056          1   
360903046    CHKN.COBB.SALAD          2.0    1.358056          1   

           meal_flow_last_to_close  total_flow_steps  meal_flow_step  \
order_id                                                               
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   
360903046                       19                20               1   

           beer_volume order_day_of_week  is_weekend  guest_count  \
order_id                                                            
360903046          0.0            Friday    

**Dec 28th (Friday), 19:25 - 20:46, 1.35 hrs, 1 meal steps.**

**Total drinks**: 1 wine

**`dinner`** 

`late night`, `1 drink` 


